### PR TITLE
Add new tutorial - Kafka version of  Count messages in a topic

### PR DESCRIPTION
**Pull request by rmoff**
This adds a `Kafka` option to the tutorial to count the number of messages in a topic (#261). 

It uses `kafkacat` because this integrates the most neatly into unix pipes. I tried `ccloud` but it doesn't have an option to exit after last message consumed, nor is there a Docker image.

### CODE REVIEW

File: _includes/tutorials/count-messages/ksql/markup/dev/03d-count-all-messages-pageview.adoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+We're specifying a `LIMIT 1` clause here so that the query exits after the first row has returned. Without this, ksqlDB will continue to write any changes in the count to the screen as new messages arrive. 
+
+You should now see a count of all the messages in the topic: 
+
++++++
+<pre class="snippet"><code class="sql">{% include_raw tutorials/count-messages/ksql/code/tutorial-steps/dev/03d-count-all-messages-pageview.out %}</code></pre>
++++++
+
+NOTE: _You may be wondering about the purpose of `X` in the query. It's a dummy field to pursuade ksqlDB to do an aggregation across all messages. You can read more about this https://github.com/confluentinc/ksql/issues/430[here]_.
```
Comment (bbejeck) [inline _includes/tutorials/count-messages/ksql/markup/dev/03d-count-all-messages-pageview.adoc:15]: For your consideration - it's not obvious from the rendered formatting that users can click on `here`, so maybe change to make it more obvious  but I don't have a better suggestion

File: _data/harnesses/count-messages/ksql.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          stdout: tutorial-steps/test/outputs/test-results.log
+          render:
+            file: tutorials/count-messages/ksql/markup/test/02-run-tests.adoc
+
+    - title: Clean up
+      content:
+        - action: execute
+          file: tutorial-steps/test/03-clean-up.sh
+          render:
+            file: tutorials/count-messages/ksql/markup/test/03-clean-up.adoc
```
Comment (bbejeck) [inline _data/harnesses/count-messages/ksql.yml:120]: I'm not sure if it fits with this tutorial but we've updated the existing files in the `harness` directory to add this extra step 
```yml
ccloud:
  steps:
    - title: Run your app to Confluent Cloud
      content:
        - action: skip
          render:
            file: shared/markup/ccloud/try-ccloud.adoc
```

File: _data/harnesses/count-messages/kafka.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        - action: execute
+          file: tutorial-steps/dev/02b-docker-compose-up.sh
+          render:
+            file: tutorials/count-messages/kafka/markup/dev/02b-start-compose.adoc
+
+        - action: execute
+          file: tutorial-steps/dev/02c-wait-for-containers.sh
+          render:
+            skip: true
```
Comment (bbejeck) [inline _data/harnesses/count-messages/kafka.yml:25]: For your consideration - render the command for waiting as you did in the ksqlDB version. 
I needed to run the `kafkacat` command twice the first time I got this 
```
ERROR: Topic pageviews error: Broker: Unknown topic or partition
       0
```
I realize it's a very minor point.

File: _includes/tutorials/count-messages/kafka/markup/dev/03a-count-messages.adoc [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+<pre class="snippet"><code class="shell">{% include_raw tutorials/count-messages/kafka/code/tutorial-steps/dev/03a-count-messages.sh %}</code></pre>
++++++
+
++++++
+<pre class="snippet"><code class="shell">{% include_raw tutorials/count-messages/kafka/code/tutorial-steps/dev/03a-count-messages.out %}</code></pre>
++++++
+
+Let's take a close look at the commandline soup we've used here to count the messages. 
+
+* `docker exec kafkacat` runs the following command with its arguments in the Docker container called `kafkacat`
```
Comment (bbejeck) [inline _includes/tutorials/count-messages/kafka/markup/dev/03a-count-messages.adoc:15]: Great description of using `kafkacat`!

**Review by bbejeck — APPROVED** [review]
Thanks for the new tutorial @rmoff.  This LGTM, I have a couple of minor comments for your consideration, but I think it can go as is if you want.

File: _includes/tutorials/count-messages/ksql/markup/dev/03d-count-all-messages-pageview.adoc (hunk 1 above)
Comment (rmoff) [reply _includes/tutorials/count-messages/ksql/markup/dev/03d-count-all-messages-pageview.adoc]: fixed. thanks.

File: _data/harnesses/count-messages/kafka.yml (hunk 3 above)
Comment (rmoff) [reply _data/harnesses/count-messages/kafka.yml]: Good catch. I've added the instruction in.

**Pull request merged**